src/coreclr/System.Private.CoreLib/src/System/Collections/Generic/ComparerHelpers.cs

```diff
@@ -122,41 +85,13 @@ internal static object CreateDefaultEqualityComparer(Type type)
             else if (type.IsEnum)
             {
                 // The equality comparer for enums is specialized to avoid boxing.
-                result = TryCreateEnumEqualityComparer(runtimeType);
-            }
-
-            return result ?? CreateInstanceForAnotherGenericParameter((RuntimeType)typeof(ObjectEqualityComparer<object>), runtimeType);
-        }
-
-        /// <summary>
-        /// Creates the default <see cref="EqualityComparer{T}"/> for an enum type.
-        /// </summary>
-        /// <param name="enumType">The enum type to create the default equality comparer for.</param>
-        private static object? TryCreateEnumEqualityComparer(RuntimeType enumType)
-        {
-            Debug.Assert(enumType != null);
-            Debug.Assert(enumType.IsEnum);
-
-            // See the METHOD__JIT_HELPERS__UNSAFE_ENUM_CAST and METHOD__JIT_HELPERS__UNSAFE_ENUM_CAST_LONG cases in getILIntrinsicImplementation
-            // for how we cast the enum types to integral values in the comparer without boxing.
 
-            TypeCode underlyingTypeCode = Type.GetTypeCode(Enum.GetUnderlyingType(enumType));
-
-            // Depending on the enum type, we need to special case the comparers so that we avoid boxing.
-            switch (underlyingTypeCode)
-            {
-                case TypeCode.Int32:
-                case TypeCode.UInt32:
-                case TypeCode.SByte:
-                case TypeCode.Byte:
-                case TypeCode.Int16:
-                case TypeCode.Int64:
-                case TypeCode.UInt64:
-                case TypeCode.UInt16:
-                    return CreateInstanceForAnotherGenericParameter((RuntimeType)typeof(EnumEqualityComparer<>), enumType);
+                // See the METHOD__JIT_HELPERS__UNSAFE_ENUM_CAST and METHOD__JIT_HELPERS__UNSAFE_ENUM_CAST_LONG cases in getILIntrinsicImplementation
```

Comment:
`METHOD__JIT_HELPERS__UNSAFE_ENUM_CAST...` does not exist anymore. This comment is no longer relevant.